hyperspy/_signals/signal2d.py

```diff
@@ -605,3 +608,433 @@ def add_ramp(self, ramp_x, ramp_y, offset=0):
+                                 scikit-image which uses a maximum filter
+                     'minmax' - finds peaks by comparing maximum filter results
+                                with minimum filter, calculates centers of mass
+                     'zaefferer' - based on gradient thresholding and refinement
+                                   by local region of interest optimisation
+                     'stat' - statistical approach requiring no free params.
+                     'massiel' - finds peaks in each direction and compares the
+                                 positions where these coincide.
+                     'laplacian_of_gaussians' - a blob finder implemented in
+                                                `scikit-image` which uses the
+                                                laplacian of Gaussian matrices
+                                                approach.
+                     'difference_of_gaussians' - a blob finder implemented in
+                                                 `scikit-image` which uses
+                                                 the difference of Gaussian
+                                                 matrices approach.
+        *args : associated with above methods
+        **kwargs : associated with above methods.
+        Returns
+        -------
+        peaks: structured array of shape _navigation_shape_in_array in which
+               each cell contains an array with dimensions (npeaks, 2) that
+               contains the x, y coordinates of peaks found in each image.
+        """
+        arr_shape = (self.axes_manager._navigation_shape_in_array
+                     if self.axes_manager.navigation_size > 0
+                     else [1, ])
+        peaks = np.zeros(arr_shape, dtype=object)
+        for z, indices in zip(self._iterate_signal(),
+                              self.axes_manager._array_indices_generator()):
+            if method == 'skimage':
+                peaks[indices] = peak_local_max(z, *args, **kwargs)
+            elif method == 'max':
+                peaks[indices] = find_peaks_max(z, **kwargs)
+            elif method == 'minmax':
+                peaks[indices] = find_peaks_minmax(z, **kwargs)
+            elif method == 'zaefferer':
+                peaks[indices] = find_peaks_zaefferer(z, **kwargs)
+            elif method == 'stat':
+                peaks[indices] = find_peaks_stat(z, **kwargs)
+            elif method == 'laplacian_of_gaussians':
+                peaks[indices] = find_peaks_log(z, **kwargs)
+            elif method == 'difference_of_gaussians':
+                peaks[indices] = find_peaks_dog(z, **kwargs)
+            else:
+                raise NotImplementedError("The method `{}` is not implemented. "
+                                          "See documentation for available "
+                                          "implementations.".format(method))
+
```

Comment:
A better approach, I feel, would be to do something like so:
```python
method_dict = {'skimage': peal_local_max,
               'max': find_peaks_max,
               'minmax': find_peaks_minmax,
               'zaefferer': find_peaks_zaefferer,
               'stat': find_peaks_stat,
               'laplacian_of_gaussians': find_peaks_log,
               'difference_of_gaussians': find_peaks_dog,
              }

if method in method_dict:
    method = method_dict[method]
else:
    raise NotImplementedError("The method `{}` is not implemented. " 
                              "See documentation for available "
                              "implementations.".format(method))
for z, indices in zip(self._iterate_signal(),
                      self.axes_manager._array_indices_generator()):
    peaks[indices] = method(z, *args, **kwargs)
```
[do check that I did not mistype this...]

I feel this makes it easier to add any new methods, and at the same time separates option-checking from the actual computation.